### DISCUSSION TITLE
Select component not passing A11y test

### DISCUSSION THREAD

**Discussion by skarensmoll** [radix-ui/primitives#2554] [Help]
Hi there!

First of all thanks a lot for all the effort and support you deliver with all your components. Recently we adopted the select component and we've got some A11y checks running however, those are complaining  because of the role combobox being applied to a button instead of an input, apparently here are the reasons:

A combobox role is generally used for an input field that allows the user either to choose an item from a predefined list or to enter a custom value. It typically consists of a textbox element and an associated listbox or similar element containing selectable options. This structure enables users, especially those using assistive technologies, to understand and interact with the component effectively.

On the other hand, a button is a simpler interactive element, generally used to perform an action when clicked, like submitting a form or opening a dialog. Assigning it a combobox role would misrepresent its function to assistive technology users, potentially leading to confusion and difficulty in navigating and using the interface.

here's the reference:

- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Combobox_Role
- https://dequeuniversity.com/rules/axe/4.2/aria-allowed-role?application=axeAPI

thanks a lot

**Comment by benoitgrelard** [comment]
A select-only combobox is a valid WAI-ARIA pattern: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/
